src/main/java/org/opensearch/security/state/SecurityMetadata.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+package org.opensearch.security.state;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.opensearch.Version;
+import org.opensearch.cluster.AbstractNamedDiffable;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.NamedDiff;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+public final class SecurityMetadata extends AbstractNamedDiffable<ClusterState.Custom> implements ClusterState.Custom {
+
+    public final static SecurityMetadata DEFAULT = new SecurityMetadata(false);
+
+    public final static String TYPE = "security";
+
+    public final static String SECURITY_CONFIGURATION_APPLIED_FIELD_NAME = "security_configuration_applied";
+
+    private final boolean securityConfigurationApplied;
```

Comment:
! This is a great, can I ask you to make a tweak that will add a feature [1] we've been looking for?

Instead of a boolean if a identifier is associated associated with it, we can make sure the security configuration is the same version on all nodes.  

Note; there is more work associated with getting the security config refresh process connected to this, but I'd be happy to do that wiring in a separate follow up pull request.

- https://github.com/opensearch-project/security/issues/3275

Reply:
WDYT to add a property `versions` so cluster state will looks like:
```json
"security": {
    "initia_security_configuration_applied": true,
    "versions": {
         "actions_group": 1,
         "roles": 1,
         ....
     }
}
```
so if a new node joins, to the cluster, we will see versions for all `CTypes` the index contains. AFAIU for the current logic, there is no way to get this info. As a result, it is possible to upgrade the `CTypes` configuration as soon as the cluster switches to the new manager with the new configuration?

Reply:
Sure - that works too